src/AutoRest.CSharp.V3/PluginProcessor.cs

```diff
@@ -1,18 +1,36 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using System.Reflection;
 using System.Text.Json;
 using System.Threading.Tasks;
 using AutoRest.CSharp.V3.CodeGen;
 using AutoRest.CSharp.V3.JsonRpc;
 using AutoRest.CSharp.V3.Pipeline;
+using AutoRest.CSharp.V3.Pipeline.Generated;
+using AutoRest.CSharp.V3.Plugins;
 using AutoRest.CSharp.V3.Utilities;
 
 namespace AutoRest.CSharp.V3
 {
     internal static class PluginProcessor
     {
-        public static string[] PluginNames { get; } = { "csharp-v3" };
+        //public static Dictionary<string, Func<IPlugin>> Plugins = new Dictionary<string, Func<IPlugin>>
+        //{
+        //    { SerializeTester.PluginName, () => new SerializeTester() },
+        //    { ModelCreator.PluginName, () => new ModelCreator() },
+        //    { SerializeTester.PluginName, () => new SerializeTester() },
+        //    { SerializeTester.PluginName, () => new SerializeTester() },
+        //};
+
+        //https://stackoverflow.com/a/26750/294804
+        private static readonly Type[] PluginTypes = Assembly.GetExecutingAssembly().GetTypes()
```

Comment:
Do we really think we will have enough plugins to warrant the need for dynamic discovery vs just a hard-coded list? Personally i don't think we will.

Reply:
It certainly makes it easier for me to work on in the short term. It is just something else I don't need to worry about until I know how this system will be designed.

Reply:
Having worked on the MEF (Managed Extensibility Framework) team I can tell you debugging dynamically loaded things is a lot harder to and so I generally only do it if I need to dynamically discover things from different assemblies. When they are all in the same assembly I would recommend statically referencing them.